src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -260,22 +264,38 @@ public static string GetFileHash(string path)
                 }
                 else
                 {
+                    installerMatchDict.Add(matchingExistingInstaller, newInstaller); // add the match to the map.
                     existingInstallers.Remove(matchingExistingInstaller);
                 }
-
-                matchingExistingInstaller.InstallerUrl = newInstaller.InstallerUrl;
-                matchingExistingInstaller.InstallerSha256 = newInstaller.InstallerSha256;
-                matchingExistingInstaller.SignatureSha256 = newInstaller.SignatureSha256;
-                matchingExistingInstaller.ProductCode = newInstaller.ProductCode;
-                matchingExistingInstaller.MinimumOSVersion = newInstaller.MinimumOSVersion;
-                matchingExistingInstaller.PackageFamilyName = newInstaller.PackageFamilyName;
-                matchingExistingInstaller.Platform = newInstaller.Platform;
             }
 
             if (unmatchedInstallers.Any() || multipleMatchedInstallers.Any())
             {
                 throw new InstallerMatchException(multipleMatchedInstallers, unmatchedInstallers);
             }
+            else
+            {
+                foreach (var existingInstaller in installerMatchDict.Keys)
+                {
+                    UpdateInstallerNode(existingInstaller, installerMatchDict[existingInstaller]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Updates the metadata from an existing installer node with the metadata from a new installer node.
+        /// </summary>
+        /// <param name="existingInstaller">Existing installer node.</param>
+        /// <param name="newInstaller">New installer node.</param>
+        public static void UpdateInstallerNode(Installer existingInstaller, Installer newInstaller)
```

Comment:
add tests for this function

Reply:
I think testing this function would be trivial as its only assigning values to the new installer. However, the new unit test I added should indirectly test this function.